File: MApps/MCtripModules/MCtripDeviceModule/MCtripDeviceAndroidModule/src/main/java/com/mctrip/modules/device/android/AndroidFragment.java
package com.mctrip.modules.device.android;

import android.annotation.SuppressLint;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.Toast;

import com.mctrip.library.base.MFragment;


public class AndroidFragment extends MFragment {
    @SuppressLint("InflateParams")
    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View contentView = inflater.inflate(R.layout.fragment_android, null);
        Button buttonAndroid = (Button) contentView.findViewById(R.id.buttonAndroid);
        buttonAndroid.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Toast.makeText(mFragmentActivity, "this is android child module", Toast.LENGTH_SHORT).show();
            }
        });
        return contentView;
    }
}

File: android/arsenal/modules/module-mobile-android/src/main/java/com.multiapk.modules.mobile.android/AndroidFragment.kt
package com.multiapk.modules.mobile.android

import android.annotation.SuppressLint
import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.Button
import com.jakewharton.rxbinding2.view.RxView
import org.smartrobot.multiapk.core.hotpatch.Hotpatch
import org.smartrobot.base.DefaultBaseFragment
import io.reactivex.android.schedulers.AndroidSchedulers
import org.jetbrains.anko.toast
import java.io.File
import java.util.concurrent.TimeUnit

class AndroidFragment : DefaultBaseFragment() {
    @SuppressLint("InflateParams")
    override fun onCreateView(inflater: LayoutInflater?, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val contentView = inflater!!.inflate(R.layout.fragment_android, null)
        val buttonAndroid = contentView.findViewById(R.id.buttonAndroid) as Button
        RxView.clicks(buttonAndroid).debounce(200, TimeUnit.MILLISECONDS).observeOn(AndroidSchedulers.mainThread()).subscribe {
            Log.w("krmao", "this is android child module" + Thread.currentThread().name)
            try {
                Hotpatch.instance.installPatch("com.multiapk.modules.mobile.ios", 1, File("/storage/emulated/0/ios.patch"))
                context?.toast("合成成功" + Thread.currentThread().name)
            } catch (e: Exception) {
                e.printStackTrace()
                context?.toast("合成失败" + Thread.currentThread().name)
            }
        }
        return contentView
    }
}
